#!/usr/bin/env python

PACKAGE = "odom"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

# For odometry
odom_1 = gen.const("Euler", int_t, 0, "Euler odometry")
odom_2 = gen.const("RK", int_t, 1, "RK odometry")
odom_enum = gen.enum([odom_1, odom_2], "Enum of methods")

# For encoder reading
enc_1 = gen.const("Ticks", int_t, 0, "Velocity from ticks data")
enc_2 = gen.const("RPM", int_t, 1, "Velocity from angular speed data")
encoder_enum = gen.enum([enc_1, enc_2], "Enum of data")

gen.add("odom_mode", int_t, 0, "Odometry method", 0, edit_method=odom_enum)
gen.add("data_read", int_t, 1, "Data for velocity calculation", 1, edit_method=encoder_enum)

exit(gen.generate(PACKAGE, "odom_omni", "parameters"))